src/pubSub/pubSub.ts

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2018-2019 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import {createInbox, JetStreamClient, NatsConnection, NatsError, StreamInfo, StringCodec} from "nats";
+import {
+    GetStreamSubjects,
+    NatsConsumerConfig,
+    NatsConsumerWiseConfigMapping,
+    NatsStreamConfig,
+    NatsStreamWiseConfigMapping,
+    NatsTopic,
+    NatsTopicMapping,
+} from "./utils";
+
+import {ConsumerOptsBuilderImpl} from "nats/lib/nats-base-client/jsconsumeropts";
+
+import {ConsumerInfo, ConsumerUpdateConfig, JetStreamManager, StreamConfig} from "nats/lib/nats-base-client/types";
+
+
+export interface PubSubService {
+    Subscribe(topic: string, callback: (msg: string) => void): void
+    updateConsumer(streamName: string, consumerName: string, existingConsumerInfo: ConsumerUpdateConfig): void
+    addOrUpdateStream(streamName: string, streamConfig: StreamConfig):void
+    checkConfigChangeReqd(existingStreamInfo: StreamConfig, toUpdateConfig: StreamConfig):Promise<boolean>
```

Comment:
remove these methods from here, these should not be exposed through this interface